plugins/logitech-hidpp/fu-logitech-hidpp-peripheral.c

```diff
@@ -611,6 +611,7 @@ fu_logitech_hidpp_peripheral_setup (FuDevice *device, GError **error)
 	if (idx != 0x00) {
 		self->is_updatable = TRUE;
 		fu_device_add_flag (FU_DEVICE (device), FWUPD_DEVICE_FLAG_IS_BOOTLOADER);
+		fu_device_set_protocol (FU_DEVICE (device), "com.logitech.unifyingsigned");
```

Comment:
this seems like an incorrect change - isn't the protocol signed only when it has idx `HIDPP_FEATURE_DFU_CONTROL_SIGNED` set?

Reply:
and furthermore the protocol was actually originally set in `fu_logitech_hidpp_peripheral_init`

Reply:
Yes, but you only get `HIDPP_FEATURE_DFU_CONTROL` or `HIDPP_FEATURE_DFU_CONTROL_SIGNED` when you're in runtime mode. If you're stuck in recovery mode (e.g. you unplugged during update) you only get HIDPP_FEATURE_DFU which doesn't tell you what protocol we should be verifying. I guess the real fix would be to change `fu_device_set_protocol()` into `fu_device_add_protocol()` but that's a lot of work for something that I don't think is very common.

Reply:
I'll split this commit out as a new PR as it's more controversial it seems.